engine/crates/operation-checks/src/check/rules.rs

```diff
@@ -160,25 +159,25 @@ pub(super) fn change_field_argument_type(
                 break 'refine; // type change overrides arity change
             }
 
-            if !src_arg.is_required && target_arg.is_required {
-                return Some(CheckDiagnostic {
-                    message: format!(
-                        "The argument `{}` became required, but clients are not providing it.",
-                        change.path
-                    ),
-                    severity: Severity::Error,
-                });
-            } else if src_arg.is_required && !target_arg.is_required {
-                return None;
+            match src_arg.wrappers.compare(&target_arg.wrappers) {
+                crate::schema::WrapperTypesComparison::BecameLessRequired => return None,
+                crate::schema::WrapperTypesComparison::BecameMoreRequired => {
```

Comment:
Why less and more required specifically? Isn't it a binary status?

Reply:
The idea of the "more"/"less" required is that `[[String]!]` is optional, but "more required" than `[[String]]`. It's not well defined, let's try to come up with something better. Maybe `AddedNonNull`/`RemovedNonNull`?